src/js/base/pagination/Pagination.tsx

```diff
@@ -53,8 +58,11 @@ export function Pagination({
         </PaginationLink>
     ));
 
-    const filters = new URLSearchParams(window.location.search);
-    filters.delete("page");
+    useEffect(() => {
+        if (currentPage > pageCount) {
+            setUrlPage(pageCount);
+        }
+    }, [items]);
```

Comment:
```suggestion
    }, [currentPage, pageCount]);
```
Does this work for the deps?